RDS module changes

alkar opened the pull request:
Connects to https://github.com/ministryofjustice/cloud-platform/issues/360

Implements a number of changes on the RDS module to bring it up to spec, as detailed in the issue above.

razvan-moj commented on README.md:
```diff
@@ -1,32 +1,32 @@
+The Terraform module will create an RDS instance, RDS subnet group and a KMS key in AWS with AWS credentials that will have access to the resources.
 
-The RDS instance created will have a name in the format of `"cloud-platform-${var.application}-${var.environment-name}-${random_string.identifier.result}"`. This ensures that the instance created is globally unique and can be recognised easily. e.g. `cloud-platform-laa-fee-calculator-staging-fjldkaw` 
+The RDS instance created will have a name in the format of `"cloud-platform-${var.application}-${var.environment-name}-${random_string.identifier.result}"`. This ensures that the instance created is globally unique and can be recognised easily. e.g. `cloud-platform-laa-fee-calculator-staging-fjldkaw`
```
not fixed length?

razvan-moj replied:
also, terraform doesn't allow dashes in name, even though https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_Limits.html says it should

alkar replied:
Good catch, the docs need updating here. I'm pretty sure dashes are allowed (after all this module works as-is). Where does it say that terraform doesn't allow dashes in the name?

razvan-moj replied:
See https://github.com/ministryofjustice/cloud-platform-environments/pull/137/files, when I was doing that I got
"* aws_db_instance.rds: Error creating DB Instance: InvalidParameterValue: DBName must begin with a letter and contain only alphanumeric characters."

alkar replied:
That's an AWS error, the `DBName` is just the name of the DB that will be created on instance launch. Our default is `<app-name><env>` which might contain dashes so we could address that too in this PR.

alkar commented:
@razvan-moj I've changed to db name as per your recommendation and updated the README and example to reflect the changes in this PR. Can you please have another look? Thanks!
